Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc

```diff
@@ -1,66 +1,104 @@
-#include "DataFormats/Alignment/interface/AlignmentClusterFlag.h"
-#include "DataFormats/Alignment/interface/AliClusterValueMap.h"
+class TrackerTopology;
+
+class AlignmentPrescaler : public edm::one::EDProducer<> {
+public:
+  AlignmentPrescaler(const edm::ParameterSet& iConfig);
+  ~AlignmentPrescaler() override;
+  void beginJob() override;
+  void endJob() override;
+  void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  // tokens
+  edm::EDGetTokenT<reco::TrackCollection> tracksToken_;   //tracks in input
+  edm::EDGetTokenT<AliClusterValueMap> aliClusterToken_;  //Hit-quality association map
 
-// #include "Riostream.h"
+  std::string prescfilename_;  //name of the file containing the TTree with the prescaling factors
+  std::string presctreename_;  //name of the  TTree with the prescaling factors
+
+  TFile* fpresc_;
+  TTree* tpresc_;
+  TRandom3* myrand_;
+
+  int layerFromId(const DetId& id, const TrackerTopology* tTopo) const;
+
+  unsigned int detid_;
+  float hitPrescFactor_, overlapPrescFactor_;
+  int totnhitspxl_;
+};
 
 using namespace std;
 
 AlignmentPrescaler::AlignmentPrescaler(const edm::ParameterSet& iConfig)
-    : src_(iConfig.getParameter<edm::InputTag>("src")),
-      srcQualityMap_(iConfig.getParameter<edm::InputTag>("assomap")),
+    : tracksToken_(consumes(iConfig.getParameter<edm::InputTag>("src"))),
+      aliClusterToken_(consumes(iConfig.getParameter<edm::InputTag>("assomap"))),
       prescfilename_(iConfig.getParameter<std::string>("PrescFileName")),
       presctreename_(iConfig.getParameter<std::string>("PrescTreeName")) {
   // issue the produce<>
   produces<AliClusterValueMap>();
   produces<AliTrackTakenClusterValueMap>();
 }
 
-AlignmentPrescaler::~AlignmentPrescaler() {
-  //
-}
+AlignmentPrescaler::~AlignmentPrescaler() = default;
 
 void AlignmentPrescaler::beginJob() {
   //
-  std::cout << "in AlignmentPrescaler::beginJob" << std::flush;
+  edm::LogPrint("AlignmentPrescaler") << "in AlignmentPrescaler::beginJob" << std::flush;
   fpresc_ = new TFile(prescfilename_.c_str(), "READ");
   tpresc_ = (TTree*)fpresc_->Get(presctreename_.c_str());
   tpresc_->BuildIndex("DetId");
   tpresc_->SetBranchStatus("*", false);
   tpresc_->SetBranchStatus("DetId", true);
   tpresc_->SetBranchStatus("PrescaleFactor", true);
   tpresc_->SetBranchStatus("PrescaleFactorOverlap", true);
-  cout << " Branches activated " << std::flush;
+  edm::LogPrint("AlignmentPrescaler") << " Branches activated " << std::flush;
   detid_ = 0;
   hitPrescFactor_ = 99.0;
   overlapPrescFactor_ = 88.0;
 
   tpresc_->SetBranchAddress("DetId", &detid_);
   tpresc_->SetBranchAddress("PrescaleFactor", &hitPrescFactor_);
   tpresc_->SetBranchAddress("PrescaleFactorOverlap", &overlapPrescFactor_);
-  cout << " addressed " << std::flush;
+  edm::LogPrint("AlignmentPrescaler") << " addressed " << std::flush;
```

Comment:
I think the `std::flush` is not needed, the `edm::LogPrint` will take care of it, no?